### PR TITLE
Indicate that Syncing FCUs are not affecting sync progress during pipeline sync

### LINKED ISSUE #7243: Indicate that `Syncing` FCUs are not affecting sync progress during pipeline sync
### Describe the feature

Currently users see the following logs when the node is syncing:
```
Mar 19 22:45:25 dan-nodes reth[2541052]: 2024-03-20T02:45:25.680926Z  INFO Status connected_peers=100 freelist=603231 stage=Execution checkpoint=6227007 target=19470460 stage_progress=8.48%
Mar 19 22:45:37 dan-nodes reth[2541052]: 2024-03-20T02:45:37.045016Z  INFO Forkchoice updated head_block_hash=0x30ffa3ffcc9741392fbdf254505248321fc378c77980241bfefae48d5fcfdf08 safe_block_hash=0x8d55f84f5624d4fe3dee8e12a1fada869a37f9e6b460cbec79935ee1990526b3 finalized_block_hash=0x76b972b20063dee91a8b76b1ddfafb2f47233af69f087f6f96a0590c3593542d status=Syncing
Mar 19 22:45:49 dan-nodes reth[2541052]: 2024-03-20T02:45:49.433627Z  INFO Forkchoice updated head_block_hash=0xed7db1017fde5c9c533029a5bd86120bf93a7753fac75f207906215216db0018 safe_block_hash=0x8d55f84f5624d4fe3dee8e12a1fada869a37f9e6b460cbec79935ee1990526b3 finalized_block_hash=0x76b972b20063dee91a8b76b1ddfafb2f47233af69f087f6f96a0590c3593542d status=Syncing
Mar 19 22:45:50 dan-nodes reth[2541052]: 2024-03-20T02:45:50.683217Z  INFO Status connected_peers=100 freelist=603231 stage=Execution checkpoint=6227007 target=19470460 stage_progress=8.48%
Mar 19 22:46:00 dan-nodes reth[2541052]: 2024-03-20T02:46:00.899792Z  INFO Forkchoice updated head_block_hash=0xee6fec6e326a9783daee533f98346b0d0fa6c036a50315d5380102a2b21052a5 safe_block_hash=0x8d55f84f5624d4fe3dee8e12a1fada869a37f9e6b460cbec79935ee1990526b3 finalized_block_hash=0x76b972b20063dee91a8b76b1ddfafb2f47233af69f087f6f96a0590c3593542d status=Syncing
```

The `Forkchoice updated` log makes it seem like the FCU affected the sync condition, but that log really just means reth received a FCU and responded `Syncing` to the CL. The FCU is likely discarded since we're in pipeline sync. We should make this more clear in the logs.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AbnerZheng**
close https://github.com/paradigmxyz/reth/issues/7243

### CODE REVIEW

**Review by Rjected — CHANGES_REQUESTED** [review]
nit: mind rebasing this on main instead of the pending session graph branch?

**Comment by AbnerZheng** [comment]
oh, sorry about it. Will do it right now.

File: crates/node-core/src/events/node.rs [hunk 1]
```diff
@@ -223,13 +223,21 @@ impl<DB> NodeState<DB> {
...
+                            "Received invalid forkchoice updated message"
+                        );
+                    } else if status == ForkchoiceStatus::Syncing {
+                        info!(
+                            ?head_block_hash,
+                            ?safe_block_hash,
+                            ?finalized_block_hash,
+                            "Received forkchoice updated message when syncing"
+                        );
+                    }
```
Comment (Rjected) [inline crates/node-core/src/events/node.rs:240]: we need to handle all cases of `ForkchoiceStatus`, should use a match for this

**Review by Rjected — CHANGES_REQUESTED** [review]

File: crates/node-core/src/events/node.rs (hunk 1 above)
Comment (AbnerZheng) [reply crates/node-core/src/events/node.rs]: Done.

File: crates/node-core/src/events/node.rs [hunk 2]
```diff
@@ -219,17 +219,28 @@ impl<DB> NodeState<DB> {
...
                 {
-                    info!(
-                        ?head_block_hash,
-                        ?safe_block_hash,
-                        ?finalized_block_hash,
-                        ?status,
-                        "Forkchoice updated"
-                    );
+                    match status {
+                        ForkchoiceStatus::Valid => {}
```
Comment (Rjected) [inline crates/node-core/src/events/node.rs:226]: we should still be sending a log here

**Review by Rjected — CHANGES_REQUESTED** [review]
one nit otherwise this looks good to me

**Review by Rjected — APPROVED** [review]
nice! this looks good to me

**Pull request merged**